src/sentry/http.py

```diff
@@ -331,16 +334,18 @@ def fetch_file(url, headers=None, domain_lock_enabled=True, outfile=None):
             # TODO(dcramer): we want to be less aggressive on disabling domains
             if domain_lock_enabled:
                 cache.set(domain_key, error or '', 300)
-            logger.warning('source.disabled', extra=error)
+                logger.warning('source.disabled', extra=error)
             raise CannotFetch(error)
 
-        if outfile is not None:
-            body = None
-        else:
-            body = b''.join(contents)
         headers = {k.lower(): v for k, v in response.headers.items()}
         encoding = response.encoding
 
+        if isinstance(outfile, file):
```

Comment:
Hmm, I'm not sure the best answer here. @mitsuhiko thoughts? This feels wrong to me.

I'd rather check for `hasattr(outfile, 'getvalue')` or something instead of checking literally if it's a subclass of `file`. Lots of things will implement a file-like interface without subclasses file itself. It's pretty pythonic to use duck typing for stuff like this.